packages/instantsearch.js/src/lib/InstantSearch.ts

```diff
@@ -594,6 +595,8 @@ See ${createDocumentationLink({
     });
 
     if (this._initialResults) {
+      hydrateSearchClient(this.client, this._initialResults);
+
       const originalScheduleSearch = this.scheduleSearch;
```

Comment:
do we need to do all this orchestration still if the client has a cache? I guess for clients without cache?

Reply:
I think it's fine to keep this for when cache is disabled indeed.